src/EditorFeatures/Core.Wpf/NavigateTo/NavigateToItemDisplay.cs

```diff
@@ -114,16 +125,15 @@ public void NavigateTo()
             //
             // In the case of a stale item, don't require that the span be in bounds of the document
             // as it exists right now.
-            //
-            // TODO: Get the platform to use and pass us an operation context, or create one
-            // ourselves.
-            _threadingContext.JoinableTaskFactory.Run(() => navigationService.TryNavigateToSpanAsync(
+            using var context = _threadOperationExecutor.BeginExecute(
+                EditorFeaturesResources.Navigating_to_definition, EditorFeaturesResources.Navigating_to_definition, allowCancellation: true, showProgress: false);
+            await navigationService.TryNavigateToSpanAsync(
                 workspace,
                 document.Id,
                 _searchResult.NavigableItem.SourceSpan,
                 NavigationOptions.Default,
                 allowInvalidSpan: _searchResult.NavigableItem.IsStale,
-                CancellationToken.None));
+                context.UserCancellationToken).ConfigureAwait(false);
```

Comment:
Not sure if that needs to be .ConfigureAwait(true) for the threaded wait dialog?

Reply:
it does not need to be CA(true).  the TWD is entirely free-threaded/bg-safe/etc.